### PR TITLE
HOTFIX: Removes all movables from charger on cyborg exit.

### LINKED ISSUE #2528: Creating rods while in cyborg charger and hands full drops rods into charger contents
Something simple minor and silly.

How to cause: Be in cyborg charger as mommi and holding a stack of metal. Attempt to create rods but less than maximum for the amount of metal you're holding. The rods will be placed inside of the contents of the charger instead of dropped outside.

### PR DESCRIPTION
**Pull request by gbasood**
Cyborg chargers now remove all movable contents when the cyborg being charged exits the charger. Resolves #2528

### CODE REVIEW

File: code/game/machinery/rechargestation.dm [hunk 1]
```diff
@@ -211,6 +211,9 @@
 	src.occupant = null
 	build_icon()
 	src.use_power = 1
+	// Removes dropped items/magically appearing mobs from the charger too
+	for (var/atom/movable/x in src.contents)
+		x.loc = src.loc
```
Comment (eswordthecat) [inline code/game/machinery/rechargestation.dm]: use x.setLoc(src.loc)

Comment (gbasood) [inline code/game/machinery/rechargestation.dm]: Would I want to change the occupant's location change to setLoc as well?

Comment (eswordthecat) [inline code/game/machinery/rechargestation.dm]: wait i'm debating with myself.

Comment (eswordthecat) [inline code/game/machinery/rechargestation.dm]: oh well just leave it at is.

**Comment by clusterfack** [comment]
Well the charger does normally HOLD objects however, for right now just power cells but in the future other potential upgrades. I think it might want to check against var/list/acceptable_upgradeables before ejecting.

**Comment by ComicIronic** [comment]
There should be a check to see if the atom is in upgrade_holder.

**Comment by clusterfack** [comment]
Yeah that's probably a superior option comic

**Comment by ComicIronic** [comment]
This is a precursor to a wider fix of crafting checking for allow_drop, which is the root of the issue. Other than that, looks good.

**Comment by gbasood** [comment]
It'd also be an issue if somehow something got dropped that wasn't crafted while inside the charger, such as if a MoMMI tried to store an item that couldn't be stored while charging, like wire.

**Pull request merged**